tests/test_util.py

```diff
@@ -0,0 +1,180 @@
+from datetime import datetime, timedelta
+
+import pytest
+
+from astropy.io import fits
+import numpy as np
+from numpy.testing import assert_array_equal
+from asdf.tags.core import Software, HistoryEntry
+
+from stdatamodels import util
+
+
+def test_gentle_asarray_array_input():
+    inp = np.array([1, 2, 3, 4], dtype=np.int16)
+    result = util.gentle_asarray(inp, dtype=np.int16)
+    assert result.dtype == np.int16
+    assert_array_equal(result, inp)
+
+    result = util.gentle_asarray(inp, dtype=np.int8)
+    assert result.dtype == np.int8
+    assert_array_equal(result, inp)
+
+    dtype = np.dtype([("col1", np.int16), ("col2", np.float32)])
+    result = util.gentle_asarray(inp, dtype=dtype)
+    assert result.dtype == dtype
+    assert result["col1"].dtype == np.int16
+    assert_array_equal(result["col1"], np.array([1, 2, 3, 4], dtype=np.int16))
+    assert result["col2"].dtype == np.float32
+    assert_array_equal(result["col1"], np.array([1, 2, 3, 4], dtype=np.float32))
```

Comment:
Yes, broadcasting out to a recarray with multiple columns from a regular array.  This hits a corner case in the `gentle_asarray` code that wasn't getting tested otherwise.  I'll add notes to this and the rest of the tests like you suggested.